### ISSUE TITLE
Cleanup API

### ISSUE THREAD

**Issue by pellared** [goyek/goyek#62]
Proposals:
1. Rename `Taskflow` type to `Flow` - shorter and lesser change to confuse with `Task`
2. Rename `TF` type to `G`  - shorter and g for goyek

**Comment by dertseha** [comment]
I'm OK with removing `Exec`.

As for renaming `TF` to `G`- two things:
1. How about also changing the parameter of the task function to an interface? (following the Go mantra of "return structs, receive interfaces"). This might also help with testing as we can't predict the evolution of this type.

2. Combined with proposed point 1 (`Taskflow` -> `Flow`), as a user, I create a `goyek.Flow` for the plan and will later receive a `goyek.G[oyek]` for an execution. With these two names, I'd have thought it the other way round: To describe the plan, I use a `goyek.Goyek`, and during execution I receive a `goyek.F[low]`. Note that I also don't propose this, as it doesn't seem to fit either; I wanted to point out that the two things seem like on two different concept levels.

I'd not use `Goyek` as basis for a type name as I see it as a project name and I prefer to have descriptive names of the things I use.
Using `Flow` as the entry-point name is OK for me, so what describes the execution of a task's `Command` in a concrete flow? The first thing that comes to mind is `Execution`, and on the path of shortening names, it would be `goyek.E`. It could also be a "TaskExecution", so `goyek.TE`.
Then I'd have calls like `t[e].Fail()`, as I am failing that particular execution of the current task.

**Comment by pellared** [comment]
> 1. How about also changing the parameter of the task function to an interface? (following the Go mantra of "return structs, receive interfaces"). This might also help with testing as we can't predict the evolution of this type.

Agree. I was thinking about it as well, some time ago. Moreover it could allow non-breaking changes in future e.g. if we allow add some plugins support.

>  The first thing that comes to mind is Execution, and on the path of shortening names, it would be goyek.E. 

I like it 👍  Also because `g` is often used for https://pkg.go.dev/golang.org/x/sync/errgroup which someone may used to introduce some concurrency within the command.

I will update the "issue description".

Just a note: I do not push to implement these changes fast. I want to group here more ideas here before we start implementing these breaking changes. Moreover, I feel that we need some time to digest the ideas. I guess we should not implement it sooner than in 2 months,

**Comment by pellared** [comment]
What about renaming:

https://github.com/goyek/goyek/blob/f5360e6abc36975bac6545ab60e7287be240a782/task.go#L19

to `Action`? 

It will reduce possible confusion with `tf.Cmd` as well as `exec.Command` from the standard package.

**Comment by pellared** [comment]
I was thinking a lot about this proposal:

> Change `TF` type 
>    1. Rename to `Progress`  - more meaningful

I think it can `TF` can stay as it does not give a lot of value and would force a lot of refactoring for existing users. I know it is easy refactoring but still, I think it is not worth the cost. Especially that once we rename `Taskflow` to `Flow` then `TF` would be a unique term.


>    2. Change from `struct` to `interface`

The thing is that the `TF` type is in reality something that is returned by `goyek`. It would be very complex to support custom `TF` implementations. For example in https://github.com/open-telemetry/opentelemery-go sometimes there are interfaces for such scenarios but they contain unexported members to not allow custom implementations. I suggest postponing this change until we have a use case or even a PR how it could help even internally.
